include/odp/api/spec/pool.h

```diff
@@ -604,6 +604,32 @@ unsigned int odp_pool_max_index(void);
  */
 int odp_pool_index(odp_pool_t pool);
 
+/**
+ * Count the number of available entries in the pool.
+ *
+ * Get the count of available entries in the pool. When the local cache is enabled,
+ * the count will account for the local cache entries per CPU as well.
+ *
+ * @param pool      Pool handle
+ *
+ * @return Return the number of available entries in the pool
+ * @retval <0     On failure (e.g., handle not valid)
+ */
+int odp_pool_available_count(odp_pool_t pool);
```

Comment:
These counts could be combined to common odp_pool_stat(pool, &stat) API. Pool statistics could contain potentially also other counters. Max utilization could be provided also. 

Application could enable / disable statistics as needed.

Reply:
I will send a separate PR for the same.